### ISSUE TITLE
a bug leads to the items missing in the trash collection.

### ISSUE THREAD

**Issue by gitlzzz** [github-young/zotero-better-authors#75]
The plugin causes the Trash collection to not show anything. Other folders are normal, but when clicking on Trash, nothing is displayed in Trash, and then this will leads to nothing being displayed in any other folder. disabling the plugin and restarting is normal, enabling it reproduces.

```
14:53:51.246 1721739231245	addons.xpi	WARN	Checking /code/Zotero_linux-x86_64/distribution/extensions for addons
14:53:51.286 NS_ERROR_ILLEGAL_VALUE: Component returned failure code: 0x80070057 (NS_ERROR_ILLEGAL_VALUE) [nsICommandLine.handleFlagWithParam] BrowserContentHandler.sys.mjs:428
14:53:51.289 NS_ERROR_ILLEGAL_VALUE: Component returned failure code: 0x80070057 (NS_ERROR_ILLEGAL_VALUE) [nsICommandLine.handleFlagWithParam]
    dch_handle resource:///modules/BrowserContentHandler.sys.mjs:1289
BrowserContentHandler.sys.mjs:1308:15
14:53:51.289 Warning: unrecognized command line flag -url
BrowserContentHandler.sys.mjs:1344:17
14:53:51.626 unreachable code after return statement util.js:205:4
14:53:51.890 1721739231890	addons.webextension.Knowledge4Zotero@windingwind.com	WARN	Loading extension 'Knowledge4Zotero@windingwind.com': Reading manifest: Warning processing version: version must be a version string consisting of at most 4 integers of at most 9 digits without leading zeros, and separated with dots
14:53:52.104 Missing resource in locale en-AU: browser/browserSets.ftl
14:53:52.106 Missing resource in locale en-CA: browser/browserSets.ftl
14:53:52.113 Missing resource in locale en-NZ: browser/browserSets.ftl
14:53:52.462 InstallTrigger is deprecated and will be removed in the future. 4 reader.js:20134:16
14:53:52.605 1721739232605	addons.xpi	WARN	Addon with ID undefined already installed, older version will be disabled
14:53:52.606 Error: Addon must include an id, version, and type 2 XPIProvider.jsm:1723:13
14:53:53.389 patching  getField betterauthors.js:451:23
14:53:53.389 patching  setField betterauthors.js:451:23
14:53:53.390 patching  isFieldOfBase betterauthors.js:451:23
14:53:53.390 patching  getField betterauthors.js:451:23
14:53:53.390 patching  setField betterauthors.js:451:23
14:53:53.390 patching  isFieldOfBase betterauthors.js:451:23
14:53:53.391 patching  getField betterauthors.js:451:23
14:53:53.391 patching  setField betterauthors.js:451:23
14:53:53.391 patching  isFieldOfBase betterauthors.js:451:23
14:53:53.391
<style id="prompt-style">
betterauthors.js:451:23
14:53:53.392
<div style="position: fixed; left: 0…ht: 100%; opacity: 0.5;">
betterauthors.js:451:23
14:53:53.393
<div style="display: none;">
betterauthors.js:451:23
14:53:53.393
<input class="prompt-input" type="text" placeholder="Select a command...">
betterauthors.js:451:23
14:53:53.393
<div class="cta">
betterauthors.js:451:23
14:53:53.393
<div class="input-container">
betterauthors.js:451:23
14:53:53.393
<div class="commands-containers">
betterauthors.js:451:23
14:53:53.393
<span class="key">
betterauthors.js:451:23
14:53:53.394
<span>
betterauthors.js:451:23
14:53:53.394
<div class="instruction">
betterauthors.js:451:23
14:53:53.394
<span class="key">
betterauthors.js:451:23
14:53:53.394
<span>
betterauthors.js:451:23
14:53:53.394
<div class="instruction">
betterauthors.js:451:23
14:53:53.394
<span class="key">
betterauthors.js:451:23
14:53:53.394
<span>
betterauthors.js:451:23
14:53:53.394
<div class="instruction">
betterauthors.js:451:23
14:53:53.394
<div class="instructions">
betterauthors.js:451:23
14:53:53.394
<div id="zotero-plugin-toolkit-prompt" class="prompt-container">
betterauthors.js:451:23
14:53:53.548 Missing resource in locale en-CA: browser/browserSets.ftl
14:53:53.558 patching  refresh betterauthors.js:451:23
14:53:53.558 patching  getField betterauthors.js:451:23
14:53:53.559 patching  setField betterauthors.js:451:23
14:53:53.559 patching  isFieldOfBase betterauthors.js:451:23
14:53:53.559 patching  getField betterauthors.js:451:23
14:53:53.559 patching  setField betterauthors.js:451:23
14:53:53.559 patching  isFieldOfBase betterauthors.js:451:23
14:53:53.559 patching  getField betterauthors.js:451:23
14:53:53.559 patching  setField betterauthors.js:451:23
14:53:53.559 patching  isFieldOfBase betterauthors.js:451:23
14:53:53.568 Missing resource in locale en-AU: browser/browserSets.ftl 2
14:53:53.568 Missing resource in locale en-CA: browser/browserSets.ftl
14:53:53.585 Missing resource in locale en-NZ: browser/browserSets.ftl
14:53:53.608 Missing resource in locale en-CA: browser/browserSets.ftl
14:53:53.612 Missing resource in locale en-NZ: browser/browserSets.ftl 2
14:53:53.687 Uncaught (in promise) undefined 2
14:53:53.694 Missing resource in locale en-AU: browser/browserSets.ftl
14:53:53.716
<div class="commands-container">
betterauthors.js:451:23
14:53:53.716 patching  viewItems BetterNotes.js:462:24
14:53:53.870 Error: Failed to load resource://services-settings/remote-settings.sys.mjs PartitioningExceptionListService.sys.mjs:88:14
14:53:53.871 Uncaught (in promise) undefined
14:53:54.778 InstallTrigger is deprecated and will be removed in the future. 8 reader.js:20134:16
14:53:55.586 This site appears to use a scroll-linked positioning effect. This may not work well with asynchronous panning; see https://firefox-source-docs.mozilla.org/performance/scroll-linked_effects.html for further details and to join the discussion on related tools and features! zoteroPane.xhtml
14:53:56.924 TypeError: item.getCreators is not a function betterauthors.js:6209:33
14:53:56.925 The above error occurred in the <ItemTree> component:

ItemTree@chrome://zotero/content/itemTree.js:88:5

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries. react-dom.js:18714:17
14:53:56.925 TypeError: item.getCreators is not a function betterauthors.js:6209:33
14:54:02.695 Missing resource in locale en-CA: devtools/client/toolbox.ftl
14:54:02.695 Missing resource in locale en-NZ: devtools/client/toolbox.ftl
14:54:02.696 Missing resource in locale en-AU: devtools/client/toolbox.ftl
```

**Comment by github-young** [comment]
Do you mean the "回收站"? Can you provide the version number of the Zotero and this plugin? I cannot reproduce this bug on my Zotero (7.0.0-beta.105+c5c613c82 (64-bit)).

**Comment by gitlzzz** [comment]
Yes, it is "回收站". The version of the plugin is 4.1.3. and Zotero 7.0.0-beta.105+c5c613c82 (64-bit).

**Comment by gitlzzz** [comment]
cool, thanks

**Comment by MickeyMiao** [comment]
7.0.0-beta.106+bf5964513, same

**Comment by github-young** [comment]
Will this bug still exist after disabling all other plugins? and,
Could you please provide the screenshot of the setting page of this plugin?

**Comment by gitlzzz** [comment]
> Will this bug still exist after disabling all other plugins? and, Could you please provide the screenshot of the setting page of this plugin? @gitlzzz @MickeyMiao thanks

Yes, it still exist even after disabling all other plugins. The screenshot is here.
![image](https://github.com/user-attachments/assets/7a92544f-2733-490c-a0cd-0c079c9848c4)

**Comment by dbcoffee** [comment]
I encountered the same issue, and I have updated to the latest version of the plugin. Everything returned to normal after disabling zotero-better-authors.

**Comment by github-young** [comment]
> I encountered the same issue, and I have updated to the latest version of the plugin. Everything returned to normal after disabling zotero-better-authors.

Could you also provide the details including Zotero version, OS version, and the settings page of this plugin? I still cannot reproduce this bug on my platform.

**Comment by jstzwj** [comment]
@gitlzzz The cause of this bug is a type annotation problem of Zotero's api in a callback function, which leads to a crash when `there are collections in the trash`. Because the type annotation of callback function parameter does not have the collection type, which is actually possible to occur, our plugin failed to handle the Collection type resulted in the crash.

Now, this bug has been fixed in the new version 4.1.4. Please install the latest version and confirm if the plugin crash issue has been resolved.

**Comment by github-young** [comment]
Thank you @jstzwj. 
@gitlzzz @dbcoffee @MickeyMiao please let me know if this issue is solve by this fix and close it if solved
